src/utils/object.js

```diff
@@ -44,3 +46,46 @@ export function map(opt_initial) {
 export function hasOwn(obj, key) {
   return hasOwn_.call(obj, key);
 }
+
+/**
+ * @param {!Object} a the destination object
+ * @param {!Object} b
+ * @param {number} currentDepth
+ * @param {number|undefined=} opt_maxDepth The maximum depth for deep merge, beyond which
+ *                               Object.assign will be used.
```

Comment:
Perhaps make this a required param `maxDepth` here and pass `opt_maxDepth || Number.POSITIVE_INFINITY` into it to be explicit.